lib/reparto.ts

```diff
@@ -3,35 +3,113 @@ import { Compi } from './compi';
 export type Tarea = {
     duracionEstimada: number;
     puntuacion: number;
-}
+};
 
-class Reparto {
+export class Reparto {
+    private _goal: number;
+    private _tareasDisponibles: Array<Tarea>;
+    private _compis: Array<Compi>;
 
-    constructor(
-        readonly compis: Array<Compi>,
-        readonly tareasDisponibles: Array<Tarea>,
-        readonly tareasCompletadas: Array<Tarea>,
-        public goal: number
-    ){}
+    constructor(readonly compis_a_asignar: Array<Compi>, readonly tareas_a_asignar: Array<Tarea>) {
+        if (compis_a_asignar.length === 0) {
+            throw new Error('El array de compañeros no puede estar vacío');
+        }
+        this._compis = new Array<Compi>();
+        this.compis_a_asignar.forEach(c => this.addCompi(c));
+
+        this._tareasDisponibles = new Array<Tarea>();
+        this.tareas_a_asignar.forEach(t => this.addTarea(t));
+
+        this._goal = this.calcularGoal(this._compis, this._tareasDisponibles);
+    }
+
+    calcularGoal(compis:Array<Compi>, tareasDisponibles:Array<Tarea>): number{
+        const totalPts = tareasDisponibles.map(tarea => tarea.puntuacion).reduce((total, pts) => total + pts, 0);
+        return totalPts/compis.length;
+    }
+
+    get goal(): number {
+        return this._goal;
+    }
+
+    asignarTareas(): Map<Compi,Tarea[]> {
+        let asignaciones = new Map<Compi, Tarea[]>();
+        this._compis.forEach(compi => asignaciones.set(compi, []));
+        // Ordenar de menor a mayor horas disponibles
+        const compisOrdenados = [...this._compis].sort((a,b) => a.horasDisponibles - b.horasDisponibles);
+        // Ordenar de mayor a menor puntuación
+        const tareasOrdenadas = [...this._tareasDisponibles].sort((a,b) => b.puntuacion - a.puntuacion);
+
+        tareasOrdenadas.forEach(tarea => {
+            const compisDisponibles = compisOrdenados.filter(compi => {
+                const tiempoAsignado = this.calcularTiempoAsignado(compi, asignaciones);
```

Comment:
Fíjate que necesitas calcular esto porque has optado por que this.compisOrdenados sea estática, y tenga las horas disponibles *al principio*. Si se recalculara cada vez que se hace una asignación y se actualizara con las horas disponibles *tras la asignación*, no necesitarías hacer esta operación.

Reply:
Corregido en: https://github.com/marinajcs/asignacionTareas/pull/24/commits/36d8fac0bf9c78b50c7b8148485b05b5901fd24d